### PR TITLE
Document provisioning steps

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- document provisioning and deprovisioning steps

**Related issue(s)**
Related: #1045

### CODE REVIEW

File: docs/kyma-environment-broker/03-03-runtime-provisioning-and-deprovisioning.md [hunk 1]
```diff
@@ -9,13 +9,33 @@ Both provisioning and deprovisioning operation consist of several steps. Each st
...
+| Provision Azure Event Hubs             | Event Hub                | Creates an Azure event-hub namespace which is a managed kakfa cluster for a kyma-runtime.                                                       | @anishj0shi      |
+| Overrides_From_Secrets_And_Config_Step | Kyma overrides           | Configures default overrides for Kyma.                                                                                                          | @jasiu001        |
+| ServiceManagerOverrides                | Service Manager          | Configures overrides with Service Manager credentials.                                                                                          | @mszostok        |
+| Request_LMS_Certificates               | LMS                      | Checks if LMS tenant is ready and request certificates. The step configures SKR fluentbit. Requires the Create_LMS_Tenant step executed before. The step does not fail Provisioning Operation. | @piotrmiskiewicz |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-03-runtime-provisioning-and-deprovisioning.md:22]:
```suggestion
| Request_LMS_Certificates               | LMS                      | Checks if the LMS tenant is ready and requests certificates. The step configures SKR Fluent Bit. It requires the Create_LMS_Tenant step to be completed before. The step does not fail the provisioning operation. | @piotrmiskiewicz |
```

Comment (klaudiagrz) [reply docs/kyma-environment-broker/03-03-runtime-provisioning-and-deprovisioning.md:22]: Let's discuss.

File: docs/kyma-environment-broker/03-03-runtime-provisioning-and-deprovisioning.md [hunk 2]
```diff
@@ -9,13 +9,33 @@ Both provisioning and deprovisioning operation consist of several steps. Each st
...
+| Create_Runtime                         | Provisioning             | Triggers provisioning in the Provisioner.                                                                                                       | @jasiu001        |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-03-runtime-provisioning-and-deprovisioning.md:23]:
```suggestion
| Create_Runtime                         | Provisioning             | Triggers provisioning of a Runtime in the Runtime Provisioner.                                                                                                       | @jasiu001        |
```

File: docs/kyma-environment-broker/03-03-runtime-provisioning-and-deprovisioning.md [hunk 3]
```diff
@@ -9,13 +9,33 @@ Both provisioning and deprovisioning operation consist of several steps. Each st
...
 
 Any deprovisioning step shouldn't block the entire deprovisioning operation. Use the `RetryOperationWithoutFail` function from the `DeprovisionOperationManager` struct to skip your step in case of retry timeout. Set at most 5min timeout for retries in your step.
 
+The deprovisioning process contains the following steps:
+| Name                         | Domain         | Status      | Description                                                                            | Owner     |
+|------------------------------|----------------|-------------|----------------------------------------------------------------------------------------|-----------|
+| Deprovision_Initialization   | Deprovisioning | Done        | Initialize DeprovisioningOperation instance with data read from ProvisioningOperation. | @jasiu001 |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-03-runtime-provisioning-and-deprovisioning.md:34]:
```suggestion
| Deprovision_Initialization   | Deprovisioning | Done        | Initialize the `DeprovisioningOperation` instance with data fetched from the `ProvisioningOperation`. | @jasiu001 |
```

File: docs/kyma-environment-broker/03-03-runtime-provisioning-and-deprovisioning.md [hunk 4]
```diff
@@ -9,13 +9,33 @@ Both provisioning and deprovisioning operation consist of several steps. Each st
...
+| Deprovision Azure Event Hubs | Event Hub      | In progress | Deletes an Azure event-hub namespace.                                                  | @montaro  |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-03-runtime-provisioning-and-deprovisioning.md:35]:
```suggestion
| Deprovision Azure Event Hubs | Event Hub      | In progress | Deletes the Azure Event Hub Namespace.                                                  | @montaro  |
```

File: docs/kyma-environment-broker/03-03-runtime-provisioning-and-deprovisioning.md [hunk 5]
```diff
@@ -9,13 +9,33 @@ Both provisioning and deprovisioning operation consist of several steps. Each st
...
+| Remove_Runtime               | Deprovisioning | Done        | Triggers deprovisioning in the Provisioner.                                            | @jasiu001 |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-03-runtime-provisioning-and-deprovisioning.md:36]:
```suggestion
| Remove_Runtime               | Deprovisioning | Done        | Triggers deprovisioning of a Runtime in the Runtime Provisioner.                                            | @jasiu001 |
```

File: docs/kyma-environment-broker/03-03-runtime-provisioning-and-deprovisioning.md [hunk 6]
```diff
@@ -9,13 +9,33 @@ Both provisioning and deprovisioning operation consist of several steps. Each st
...
+The provisioning process contains the following steps:
+| Name                                   | Domain                   | Description                                                                                                                                     | Owner            |
+|----------------------------------------|--------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------|------------------|
+| Initialisation                         | Provisioning             | Starts the provisioning process and asks the Director for dashboard_url if the provisioning in Gardener is done.                                | @jasiu001        |
+| Resolve_Target_Secret                  | Hyperscaler Account Pool | Resolves the name of gardener secret with Hypescaler account credentials to be used during cluster provisioning.                                | @koala7659       |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-03-runtime-provisioning-and-deprovisioning.md:16]:
```suggestion
| Resolve_Target_Secret                  | Hyperscaler Account Pool | Provides the name of a Gardener Secret that contains  Hypescaler account credentials used during cluster provisioning.                                | @koala7659       |
```

**Comment by piotrmiskiewicz** [comment]
Currently, the LMS is not required, but it will be changed soon. See https://github.com/kyma-incubator/compass/issues/1178 After the issue will be solved, all steps will be required in the production environment.

**Comment by piotrmiskiewicz** [comment]
There is only one dependency between steps, which is in the description. I've decided to not create a dedicated column for that.

**Pull request merged**